app/scripts/metamask-controller.js

```diff
@@ -2456,18 +2456,21 @@ export default class MetamaskController extends EventEmitter {
    * array if no accounts are permitted.
    *
    * @param {string} origin - The origin whose exposed accounts to retrieve.
+   * @param {boolean} suppressUnauthorized - suppresses the unauthorized error. defaults to `true`.
    * @returns {Promise<string[]>} The origin's permitted accounts, or an empty
    * array.
    */
-  async getPermittedAccounts(origin) {
+  async getPermittedAccounts(origin, suppressUnauthorized = true) {
     try {
       return await this.permissionController.executeRestrictedMethod(
         origin,
         RestrictedMethods.eth_accounts,
       );
     } catch (error) {
-      if (error.code === rpcErrorCodes.provider.unauthorized) {
-        return [];
+      if (suppressUnauthorized) {
+        if (error.code === rpcErrorCodes.provider.unauthorized) {
+          return [];
```

Comment:
should we throw a different error in case `error.code === rpcErrorCodes.provider.unauthorized` is not true?

Reply:
it needs to keep the original behaviour to not break other uses of `getAccounts` like when you call `eth_accounts` without permissions it needs to return `[]` for legacy reasons.